### PR TITLE
FlatText is a private class

### PR DESCRIPTION
**Pull request by Morriar**
This PR does two things:
* Define `FlatText` as a private class. Since every clients were using an intrude import, their code do not change.
* Remove nullable warning in text::flat

### CODE REVIEW

**Comment by Morriar** [comment]
@R4PaSs you should check this.

File: lib/core/text/flat.nit [hunk 1]
```diff
@@ -1148,8 +1148,9 @@ redef class Array[E]
 	do
 		var l = length
 		if l == 0 then return ""
-		if l == 1 then if self[0] == null then return "" else return self[0].to_s
-		var its = _items
+		var first = self[0]
+		if l == 1 then if first == null then return "" else return first.to_s
+		var its = _items.as(not null)
```
Comment (R4PaSs) [inline lib/core/text/flat.nit]: Come to think of it, having its declared before the ifs, and replacing the `self[0]` by `its[0]` could be a small optimization :)

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
+1

File: lib/core/text/flat.nit (hunk 1 above)
Comment (Morriar) [inline lib/core/text/flat.nit]: Done

**Pull request merged**